checkpatch_hook/__init__.py

```diff
@@ -0,0 +1,262 @@
+            if commit_patch:
+                _run_checkpatch(commit_patch, errors, config_dir_str, **post_dconfig)
+            else:
+                logger.debug(f"No changes to check in {config_dir}")
+        else:
+            logger.debug(f"No files to check in {config_dir}")
+
+    # print errors
+    for filename, f_errors in errors.items():
+        logger.error(f"{filename}:")
+        for error in f_errors:
+            logger.error(f'  {error["line"]}: {error["message"]}')
+    if errors:
+        sys.exit(1)
+
+
+def _get_commit_files_from_log() -> list[Path]:
+    cmd = ["git", "diff-tree", "--no-commit-id", "--name-only", "HEAD", "-r"]
+    log_output = check_output(cmd).decode().rstrip("\n")
+    return [Path(file_path) for file_path in log_output.split("\n")]
+
+
+def _get_patch(running_in_ci: bool, patch_files: list[Path]) -> str:
+    logger.debug(f"Running in CI: {running_in_ci}, patch files: {patch_files}")
+    if running_in_ci:
+        return check_output(
+            [
+                "git",
+                "format-patch",
+                "--stdout",
+                "-1",
+                "HEAD",
+                "--",
+                *patch_files,
+            ]
+        ).decode()
+    commit_msg = _read_commit_msg_from_file()
+    return _diff_to_patch(patch_files, commit_msg)
+
+
+def _diff_to_patch(patch_files: list[Path], commit_msg: str) -> str:
+    diff = check_output(["git", "diff", "--cached", "--", *patch_files]).decode()
+    # No changes, no patch
+    if not diff:
+        return ""
+    return PATCH_TEMPLATE.format(
+        diff=diff,
+        commit_msg=commit_msg,
+        current_datetime=datetime.now().strftime("%a, %d %b %Y %H:%M:%S %z"),
+    )
+
+
+def _read_commit_msg_from_file() -> str:
+    commit_msg_file = Path(".git") / "COMMIT_EDITMSG"
+    return commit_msg_file.read_text() if commit_msg_file.exists() else ""
+
+
+def _run_checkpatch(
+    commit_patch: str,
+    errors: DefaultDict[str, list],
+    config_dir_str: str,
+    errors_enabled: list | None = None,
+    errors_ignored: list | None = None,
+    max_line_length: int | None = None,
+) -> None:
+    cmd = [
+        "checkpatch.pl",
+        "--strict",  # Be more annoying
+        "--no-tree",  # No kernel source tree present
+        # Some stuff to make parsing easier (see below):
+        "--emacs",
+        "--terse",
+        "--showfile",
+        "--no-summary",
+        "--color=never",
+        "--no-signoff",  # Don't require Signed-Off-By
+    ]
+
+    if errors_ignored is not None and errors_ignored:
+        cmd += ["--ignore", ",".join(errors_ignored)]
+    elif errors_enabled is not None and errors_enabled:
+        cmd += ["--types", ",".join(errors_enabled)]
+    if max_line_length is not None:
+        cmd += [f"--max-line-length={max_line_length}"]
+
+    logger.debug(f"Running checkpatch for \"{config_dir_str}\" directory with: {' '.join(cmd)}")
```

Comment:
```suggestion
    logger.debug(f"Running checkpatch in {config_dir_str} with: {' '.join(cmd)}")
```
Would be consistent with the other log message and the wording is slightly more accurate.